web/modules/custom/custom_page/src/Controller/CustomPageController.php

```diff
@@ -59,7 +56,28 @@ public function createCustomPageAccess(RdfInterface $rdf_entity) {
       return AccessResult::allowed();
     }
 
+    // Check if the user has the OG permission to create a custom page.
+    if (OgAccess::userAccessGroupContentEntityCrud('create', $rdf_entity, $this->createNewCustomPage($rdf_entity))->isAllowed()) {
```

Comment:
This is not something that needs fixed as we will not need it in our project, but there is an actual security gap here.
If another rdf_entity is a group and not intended for having custom pages, but the access is accidentally assigned to that user, he will be able to create a custom page there. This is because the `if ($rdf_entity->bundle == 'collection' ` above is part of the global permission check and if it does not guarantee that this page _is_ a collection page.